src/Uno.UI/UI/Xaml/UIElement.skia.cs

```diff
@@ -327,21 +327,15 @@ internal virtual void OnArrangeVisual(Rect rect, Rect? clip)
 
 			_renderTransform?.UpdateFlowDirectionTransform();
 
-			// The clipping applied by our parent due to layout constraints are pushed to the visual through the ViewBox property
-			// This allows special handling of this clipping by the compositor (cf. ShapeVisual.Render).
+			// The clipping applied by our parent due to layout constraints are pushed to the visual through the ClipRectFromArrangeCycle property
+			// This allows special handling of this clipping by the compositor (cf. ContainerVisual.Render).
 			if (clip is null)
 			{
-				visual.ViewBox = null;
```

Comment:
ViewBox is a public API. We should make sure about the behavior on WinUI.

Reply:
I've tested a bunch of cases and ViewBox seems to not be affected at all. ViewBox seems to have a very specific use case: mapping the visual to new bounds (e.g. drawing in a 200x200 rect but actually mapping the result to be in a 20x20 rectangle). That wouldn't be very compatible with the layout clipping set during the arrange cycle.

Reply:
@ramezgerges Just a note, IIRC, sometimes you need to wait for the next frame after you have called `ElementCompositionPreview.GetElementVisual` to find properties initialized. I guess you might find `Size` initialized. (and from my testing it wasn't that the values are simply delayed by one frame, but really depends on when the first call to GetElementVisual was made)